docs/hooks.md

```diff
@@ -59,7 +59,22 @@ This is used in SystemJS core to ensure that import maps are loaded so that the
 
 This function downloads and executes the code for a module. The promise must resolve with a "register" array, as described in the `getRegister` documentation.
 
-The default system.js implementation is to append a script tag that downloads and executes the module's code, subsequently resolving the promise with the most recent register: `resolve(System.getRegister())`. [Example](https://github.com/systemjs/systemjs/blob/master/src/features/script-load.js).
+The default system.js implementation is to append a script tag that downloads and executes the module's code, subsequently resolving the promise with the most recent register: `resolve(System.getRegister())`. 
+
+It is impossible to have an "identifier" to mark the module identity like AMD therefore SystemJS is using a race condition prone way to register the module. Therefore, you must make sure to maintain the execution order of the code.
+
+The correct execution order looks like this:
+
+```
+async System.instantiate
+  => (The module calls) System.register
```

Comment:
Perhaps clarify this is the script tag, and the primary registration call.

Reply:
I'm describing a general process of module registration, which might have no script tag is available.

Reply:
Perhaps indicate that `System.instantiate` is triggered by the import operation first?

Perhaps use a real module example with a .js extension, then replace `(The module calls)` with `module.js:` or similar.

Reply:
Rewrite with a concreate example